tests/unit-tests/mailpoet/test-class-sensei-mailpoet.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * File with class for testing Sensei MailPoet.
+ *
+ * @package sensei-tests
+ */
+
+// phpcs:disable WordPress.NamingConventions.ValidFunctionName.MethodNameInvalid -- Mocking an external library.
+// phpcs:disable Generic.Files.OneObjectStructurePerFile.MultipleFound -- Using PHPUnit conventions.
+
+/**
+ * Class for testing Sensei_MailPoet class.
+ *
+ * @group Sensei MailPoet
+ */
+class Sensei_MailPoet_Test extends WP_UnitTestCase {
+	use Sensei_Course_Enrolment_Manual_Test_Helpers;
+	/**
+	 * Factory object.
+	 *
+	 * @var Sensei_Factory
+	 */
+	protected $factory;
+
+	/**
+	 * Set up the test.
+	 */
+	public function setUp(): void {
+		parent::setUp();
+
+		$this->factory = new Sensei_Factory();
+		$mailpoet_api  = Sensei_MailPoetAPIMockFactory::MP();
+		new Sensei_MailPoet( $mailpoet_api );
+	}
+
+	/**
+	 * Tests manual enrolment, we are not testing anything specific to the MailPoet API.
+	 * This general test ensures the hook 'sensei_manual_enrolment_learner_enrolled' throws no error in the MailPoet class.
+	 */
+	public function testEnrolStudentNoErrors() {
+		$provider   = $this->getManualEnrolmentProvider();
+		$course_id  = $this->factory->course->create();
+		$student_id = $this->factory->user->create();
+
+		$this->assertFalse( $provider->is_enrolled( $student_id, $course_id ), 'Student should not be enrolled before we enrol them.' );
+
+		$provider->enrol_learner( $student_id, $course_id );
+
+		$this->assertTrue( $provider->is_enrolled( $student_id, $course_id ), 'Student should now be enrolled without any errors from the MailPoet class.' );
+	}
+
+	/**
+	 * Tests manual unenrolment, we are not testing anything specific to the MailPoet API.
+	 * This general test ensures the hook 'sensei_manual_enrolment_learner_withdrawn' throws no error in the MailPoet class.
+	 */
+	public function testWithdrawStudentNoErrors() {
+		$provider   = $this->getManualEnrolmentProvider();
+		$course_id  = $this->factory->course->create();
+		$student_id = $this->factory->user->create();
+
+		$this->directlyEnrolStudent( $student_id, $course_id );
+		$this->assertTrue( $provider->is_enrolled( $student_id, $course_id ), 'Student should be enrolled after directly enrolling them.' );
+
+		$provider->withdraw_learner( $student_id, $course_id );
+
+		$this->assertFalse( $provider->is_enrolled( $student_id, $course_id ), 'Student should not be enrolled after withdrawing them from the course.' );
+	}
+
+}
+
+/**
+ * Stub to instantiate the MailPoet API object.
+ *
+ * @since $$next-version$$
+ */
+class Sensei_MailPoetAPIMockFactory {
+	/**
+	 * Mock MP static method.
+	 */
+	public static function MP() {
+		return new Sensei_MailPoetMockAPI_Test();
+	}
+}
```

Comment:
Could we move this (and other classes in this file) to separate files? This will help other people re-use the same logic when they want to test it.